### PR TITLE
GKE: Fix rate limiter to work on urllib urlopen

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/WQ5UcGlf/2580-gke-fix-rate-limiter-to-work-on-urllib-urlopen

api client uses urllib under the hood with default retry configuration, which gives you 10 retries whenever request fails, which clogs up rate limiter on GKe side.
We need to fix rate limiter to make urllib wait too and reduce retry count to 3. 

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/utils/k8s.py [hunk 1]
```diff
@@ -89,14 +89,31 @@ def run(self) -> None:
...
-        client._ApiClient__call_api = call_api  # deepcode ignore W0212: monkey patching in here
+        # deepcode ignore W0212: monkey patching in here
+        orig_connection_from_host = client.rest_client.pool_manager.__class__.connection_from_host
+        orig_pm_urlopen = client.rest_client.pool_manager.__class__.urlopen
+
+        @wraps(orig_pm_urlopen)
+        def pm_urlopen(*args1, **kwargs1):
+            kwargs1['retries'] = 2
+            return orig_pm_urlopen(*args1, **kwargs1)
+
```
Comment (enaydanov) [inline sdcm/utils/k8s.py:100]: Number of retries is configurable thru k8s `Configuration` object: https://github.com/kubernetes-client/python/blob/eb9b6d609e44838f4ee04477d9908d581481ec72/kubernetes/client/configuration.py#L165. Don't need to change it this way.

**Review by enaydanov — CHANGES_REQUESTED** [review]
Don't like idea to increase usage of monkey patching. It was used as a quick'n'dirty solution to make a PoC, but if we'll go further with limiting the API call rate, we need to implement it in better way. What about making own subclass of ApiClient?

But for now, I think, we can set retry number to 1 using Configuration and don't go deeper to this rabbit's hole.

Also there is an option to use custom Retry object with own `sleep()` method which wait for API call:

https://github.com/urllib3/urllib3/blob/83ddac1622426f5847bc2cee95f66ec0cf367604/src/urllib3/util/retry.py#L390

File: sdcm/utils/k8s.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/utils/k8s.py:100]: This is recent change, does not preset in version we are currently using.

**Comment by dkropachev** [comment]
> Don't like idea to increase usage of monkey patching. It was used as a quick'n'dirty solution to make a PoC, but if we'll go further with limiting the API call rate, we need to implement it in better way. What about making own subclass of ApiClient?
> 
> But for now, I think, we can set retry number to 1 using Configuration and don't go deeper to this rabbit's hole.
> 
> Also there is an option to use custom Retry object with own `sleep()` method which wait for API call:
> 
> https://github.com/urllib3/urllib3/blob/83ddac1622426f5847bc2cee95f66ec0cf367604/src/urllib3/util/retry.py#L390

I don't like it too, but unfortunately there is no other way to get there, at least on current version of kubernetes-client.
Also setting retry to 1 is bad idea, when we interact with API we always assume there there are retries in there.
Making API call failing with first try will massively increase false positives.

File: sdcm/utils/k8s.py (hunk 1 above)
Comment (enaydanov) [reply sdcm/utils/k8s.py:100]: So, we need to bump kubernetes-py library version?

Comment (dkropachev) [reply sdcm/utils/k8s.py]: done, no monkeypatching anymore

**Review by bentsi — APPROVED** [review]

**Pull request merged**